### PR TITLE
Wrong language in emails

### LINKED ISSUE #1495: Emails are always in the bookers language
Found a bug? Please fill out the sections below. 👍

### Issue Summary

Right now, the email that is being sent to both receiver and booker are always in the bookers language.

The emails need to be separate languages: the language of the booker and the language of the receiver. If no language is detected on the client, default to English

### Steps to Reproduce

1. change your user agent to another language
2. book your own link

### PR DESCRIPTION
**Pull request by alishaz-polymath**
## What does this PR do?

- Adds language to `Person` which contains : 1. translationFunction (tFunction) and 2. locale ("en","fr", etc.)
- Fixes Email language so that the attendee gets email in their language while the organizer gets the email in the language selected in their profile
- Adds a column in Attendee to hold attendee's locale so that even on events requiring confirmation, the attendees can get emails in their locale/language

## What's missing/follow up?

- The guests will currently get email in booker's language which isn't ideal (should fall back to "en") but the current code doesn't allow it. Currently figuring out if that can be taken care of in this PR itself or a follow-up would be better.
 

Fixes #1495 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

- [x] In the staging deployment, change the language for Rick Example to one language and then book a slot from another browser (to avoid session using the same locale), enter the details as `rick@example.com` while booking as well. The received emails accessible at Mailtrap should be in the respective languages

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code and corrected any misspellings
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by PeerRich** [comment]
what's required to go live with this?

**Comment by alishaz-polymath** [comment]
> what's required to go live with this?

I'm testing things out here as I am unable to test this locally. If I'm right, I should be able to tweak things a bit and get this working today. The deployment was failing yesterday because of the DB server not responding.

File: lib/CalEventParser.ts [hunk 1]
```diff
@@ -29,26 +29,26 @@ export const getWho = (calEvent: CalendarEvent) => {
...
-${attendee?.name || calEvent.language("guest")}
+${attendee?.name || calEvent.organizerLanguage("guest")}
 ${attendee.email}
       `;
     })
     .join("");
 
   const organizer = `
-${calEvent.organizer.name} - ${calEvent.language("organizer")}
+${calEvent.organizer.name} - ${calEvent.organizerLanguage("organizer")}
```
Comment (emrysal) [inline lib/CalEventParser.ts:39]: calEvent.organizer.language

**Review by emrysal — DISMISSED** [review]
So I think (Sorry, major change here) we should make "language" a property of `Person`. 

```
Person {
  name,
  email,
  timeZone,
  language
}
```

I also think this language should be a string ("en", "fr"). Something of the calEvent we can store in the Booking. The language function should be obtained otherwise. Seperate from calEvent I think.

**Review by zomars — COMMENTED** [review]
Agreed with @emrysal. We should strive for more maintainable code instead of constantly patching everywhere.

**Review by zomars — APPROVED** [review]
Code looking good and tested on preview. Let's merge this and monitor any edge cases.

**Pull request merged**